### PR TITLE
Improve static typing in `bokeh.command`

### PR DESCRIPTION
**Pull request by mattpap**

### CODE REVIEW

**Comment by mattpap** [comment]
@bryevdv, can you explain this:
```py
E         --- /home/runner/work/bokeh/bokeh/tests/unit/bokeh/util/test_dataclasses.py:before	2021-05-01 10:35:37.754999
E         +++ /home/runner/work/bokeh/bokeh/tests/unit/bokeh/util/test_dataclasses.py:after	2021-05-01 10:43:01.485462
E         @@ -10,14 +10,15 @@
E          #-----------------------------------------------------------------------------
E          from __future__ import annotations
E          
E         +# Standard library imports
E         +from typing import List, Optional, Union
E         +
E          import pytest ; pytest
E          
E          #-----------------------------------------------------------------------------
E          # Imports
E          #-----------------------------------------------------------------------------
E          
E         -# Standard library imports
E         -from typing import List, Optional, Union
E          
E          # Module under test
E          import bokeh.util.dataclasses as dc # isort:skip
```

File: bokeh/util/dataclasses.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+class _UnspecifiedType:
+    def __repr__(self) -> str:
+        return "Unspecified"
+
+Unspecified = _UnspecifiedType()
+
+_T = TypeVar("_T")
+NotRequired = Union[_UnspecifiedType, _T]
+
+def values(obj: Any) -> Iterable[Tuple[Field[Any], Any]]:
```
Comment (mattpap) [inline bokeh/util/dataclasses.py:66]: Note, there's `dataclasses.asdict()`, but it works recursively and more importantly it deep copies (!) leaf values.

**Comment by bryevdv** [comment]
@mattpap  the boilerplate purest import is missing the isort:skip comment

File: bokeh/command/subcommand.py [hunk 2]
```diff
@@ -112,10 +143,11 @@ def __init__(self, parser: ArgumentParser) -> None:
         self.parser = parser
         args = getattr(self, 'args', ())
         for arg in args:
-            flags = arg[0]
+            flags, spec = arg
             if not isinstance(flags, tuple):
                 flags = (flags,)
-            self.parser.add_argument(*flags, **arg[1])
+            spec = { field.name: value for field, value in values(spec) }
```
Comment (bryevdv) [inline bokeh/command/subcommand.py:149]: This could just be `spec.__dict__`

Comment (bryevdv) [reply bokeh/command/subcommand.py:149]: Oh, I guess you are also filtering, but then `values` should be `defined_values` or something that indicates that (or make a helper func that returns the dict and make things cleaner in this spot)

**Pull request merged**